# Exercise 2.2

In [1]:
import numpy as np

First, we draw the samples for each child. Concerning the sex, 1 represents a boy and 0 a girl. As for the day of birth, we draw integer numbers between 0 and 6, where 0 represents Monday, 1 Tuesday and so on. For the date of birth, we analogously draw numbers between 0 and 365 where 0 represents 1st of January, 2 represents 2nd of January and so on.

In [36]:
np.random.seed(43)
N = 10000000
[sex1, sex2] = np.random.randint(0,2,size=(2,N))
[day1, day2] = np.random.randint(0,7,size=(2,N))
[date1, date2] = np.random.randint(0,365,size=(2,N))

Case 1: No additional information

In [37]:
p = np.sum(sex1 * sex2) / N
print(p)

0.2499826


The rate of (boy,boy) agrees with the probability derived in lecture ($p_1=1/4$)

Case 2: Create a subset, where at least one of the children is a boy.

In [38]:
subset1 = sex1[(sex1==1) + (sex2==1)]
subset2 = sex2[(sex1==1) + (sex2==1)]
p = np.sum(subset1 * subset2) / len(subset1)
print(p)

0.333281160091


Agrees with $p_2 = 1/3$.

Case 3: The older child (child1) is boy

In [39]:
subset = sex2[sex1==1]
p = np.sum(subset) / len(subset)
print(p)

0.499941102839


Agrees with $p_3=1/2$

Case 4: Sunday corresponds to day=6 in our data. So create a subset where at least one of the children is a boy and has birthday on sunday.

In [40]:
subset1 = sex1[((sex1==1) * (day1==6)) + ((sex2==1) * (day2==6))]
subset2 = sex2[((sex1==1) * (day1==6)) + ((sex2==1) * (day2==6))]
p = np.sum(subset1 * subset2) / len(subset1)
print(p)

0.481613426119


Agrees with $p_4=\frac{2\cdot 7-1}{4\cdot 7-1}=13/27\approx 0.48148$

Case 5: Let's define "today" as the 150th day of the year corresponding to date=149 in our date. So create a subset where at least one of the children is a boy and has birthday today.

In [44]:
date = 150
subset1 = sex1[((sex1==1) * (date1==date)) + ((sex2==1) * (date_2==date))]
subset2 = sex2[((sex1==1) * (date1==date)) + ((sex2==1) * (date_2==date))]
p = np.sum(subset1 * subset2) / len(subset1)
print(p)

0.502396514161


Approximately agrees with $p_5=\frac{2\cdot 365-1}{4\cdot 365-1}=729/1459\approx 0.499657$, but the accuracy depends on the chosen day. For date=150 for example, our sampled probability is larger than 1/2 ($\approx 0.5024). Thus, we would have to further increase the overall sample size to obtain more accurate results also in this case.